_pytest/python.py

```diff
@@ -293,11 +293,11 @@ def pytest_pycollect_makeitem(collector, name, obj):
     elif collector.istestfunction(obj, name):
         # mock seems to store unbound methods (issue473), normalize it
         obj = getattr(obj, "__func__", obj)
-        if not isfunction(obj):
+        if not isfunction(get_real_func(obj)):
             collector.warn(code="C2", message=
                 "cannot collect %r because it is not a function."
                 % name, )
-        if getattr(obj, "__test__", True):
+        elif getattr(obj, "__test__", True):
```

Comment:
@mbirtwell if you keep the `if` instead of replacing it with an `elif`, does your test still passes? I ask because it seems to me that we should try to consider any object which has an explicit `__test__` attribute.